theano/gof/cc.py

```diff
@@ -306,13 +307,15 @@ def get_c_init(r, name, sub):
 
 def get_c_extract(r, name, sub):
     """Wrapper around c_extract that initializes py_name from storage."""
-    if any([getattr(c.op, 'check_input', config.check_input) for (c, _) in
-            r.clients]):
+    if any([getattr(c.op, 'check_input', config.check_input)
+            for (c, _) in r.clients
+            if not isinstance(c, string_types)]):
```

Comment:
Do you understand why there is no check like `r.owner and getattr(r.owner.op, 'check_input', True)` like in get_c_declare? This could cause some check to be skiped.

I think it would be better to comment why this is like this. It took both of us some times to understand what it need to be.